src-built-in/components/advancedAppCatalog/src/stores/storeActions.js

```diff
@@ -235,8 +235,28 @@ async function addApp(id, cb = Function.prototype) {
 		delete appConfig.window.url;
 	}
 
-	if (typeof app.manifest !== "object") {
-		appConfig.manifest = { ...appConfig };
+	let manifest;
+	if (app.manifestType === "Finsemble") {
+		try {
+			//Attempt to parse a string as JSON
+			manifest = JSON.parse(app.manifest);
+		} catch(e) {
+			try {
+				//If parsing fails, assume the manifest is a valid url that will return JSON
+				const urlRes = await fetch(app.manifest, { method: "GET" });
+
+				//Attempt to parse the url response
+				manifest = await urlRes.json();
+			} catch(e) {
+				FSBL.Clients.Logger.system.warn(`${app.title || app.name} is missing a valid manifest or URI that delivers a valid JSON manifest. Unable to add app`);
+				return cb();
```

Comment:
Here and on 259 there is an error state yet the callback is being executed with no context of an error. *Something* should be passed up to let the caller know this failed.

Reply:
So, just mentioned this one to Joe on Slack. This is a pre-existing condition callback condition. As much as it is terrible to keep letting it happen, right now we shouldn't hold this up on the API being crud.